packages/core/src/textures/TextureSystem.ts

```diff
@@ -108,16 +121,16 @@ export class TextureSystem extends System
      *
      * If you want to unbind something, please use `unbind(texture)` instead of `bind(null, textureLocation)`
      *
-     * @param {PIXI.Texture|PIXI.BaseTexture} texture - Texture to bind
+     * @param {PIXI.Texture|PIXI.BaseTexture} texture_ - Texture to bind
      * @param {number} [location=0] - Location to bind at
      */
-    bind(texture, location = 0)
+    bind(texture: Texture|BaseTexture, location = 0): void
     {
         const { gl } = this;
 
         if (texture)
         {
-            texture = texture.baseTexture || texture;
+            texture = texture.castToBaseTexture();
```

Comment:
Here lays the problem: I found in my custom pixi fork that in some shaders texture locations had numbers in them, obviously, just a sampler location number. Maybe we'll stumble across it in PixiJS apps too, if someone made filters that have default uniform with location number instead of texture type. I suggest we check if its actually texture, maybe with `if (texture && texture.castToBaseTexture) { ... `

@GoodBoyDigital

Reply:
Resolved, Mat said "let it throw"